### PR TITLE
[3] Setup flavors for iOS

### LINKED ISSUE #3: As a user, I can run the project in different flavors
## What happened
- Having different app variants/flavors is essential to allow flexible configuration in between staging/production environments.

## Acceptance Criteria:
- We must have 2 variants of setup as default:
  - Staging
  - Production
- The variants will be configured in both iOS & Android cores.
- At this stage, we will simply allow each variant to provide a different key as  Proof Of Work. For example, we expose a `String secret` that can be configured to be different in Staging/Production - and they behave the same in iOS/Android.
- **Note**: avoid using `flavorizr` or if we use it, we must exclude it from the dependency structure, as well as making proper changes to match with our convention. The other important point is that: if the app is running in `staging`, then it must NOT include any pieces of configured data from `production`; and vice versa.

### PR DESCRIPTION
**Pull request by manh-t**
Close #3

## What happened 👀

Having different app variants/flavors is essential to allow flexible configuration in between staging/production environments.
 
## Insight 📝

This part is for iOS. 🙏

- Setup 2 variants of setup as default:
  - Staging
  - Production
- Config the app name for each flavor.
- Config build actions to provide the keys that can be configured to be different in Staging/Production.
 
## Proof Of Work 📹

https://user-images.githubusercontent.com/60863885/130412182-d0d6b9a1-e964-47a2-908a-fe608b5039b3.mp4

### CODE REVIEW

**Review by luongvo — APPROVED** [review]
lgtm

File: ios/Runner.xcodeproj/xcshareddata/xcschemes/production.xcscheme [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                     BlueprintName = "Runner"
+                     ReferencedContainer = "container:Runner.xcodeproj">
+                  </BuildableReference>
+               </EnvironmentBuildable>
+            </ActionContent>
+         </ExecutionAction>
+         <ExecutionAction
+            ActionType = "Xcode.IDEStandardExecutionActionsCore.ExecutionActionType.ShellScriptAction">
+            <ActionContent
+               title = "Run Script"
```
Comment (markgravity) [inline ios/Runner.xcodeproj/xcshareddata/xcschemes/production.xcscheme:28]: You can describe this script by renaming this title.
Screen Shot 2021-08-24 at 10 39 21

Comment (manh-t) [reply ios/Runner.xcodeproj/xcshareddata/xcschemes/production.xcscheme:28]: @markgravity Thanks for the suggestion. Will update 🙏

**Review by markgravity — APPROVED** [review]
LGTM

File: ios/Runner.xcodeproj/xcshareddata/xcschemes/staging.xcscheme [hunk 2]
```diff
@@ -0,0 +1,112 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (luongvo) [reply ios/Runner.xcodeproj/xcshareddata/xcschemes/staging.xcscheme:1]: @sleepylee you skipped this comment?

**Pull request merged**